lib/rubocop/cop/style/method_defined.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
```

Comment:
Alternatively, this could be in Performance department, but I think that `method_defined?` is also better style, in addition to the performance benefit.

Reply:
Personal opinion: This clearly improves performance but I would not want it in `rubocop-performance`. I see that project for cops that makes code more ugly, harder to understand, or unsafe. That's not how it always is in reality, but I wish it was so.

Like you said, it also makes it more concise so I would say it fits.

Reply:
I came up with this concept as well when I saw https://github.com/ruby/prism/pull/3714. Since it doesn't sacrifice readability for performance, so `Style` department seems like the appropriate place for it.

I will consider whether there is a better cop name than `Style/ModuleMemberPredicate` and will look into the implementation details later.

Reply:
I think a name like `ModuleMemberExistanceCheck` is a bit more descriptive.